pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -337,3 +375,52 @@ func gitRefInfo(repo git.Repository, dir, ref string) (buildapi.GitRefInfo, erro
 	info.Message = lines[4]
 	return info, nil
 }
+
+// WaitForBuildComplete waits for a build identified by the name to complete
+func WaitForBuildComplete(c osclient.BuildInterface, name string) error {
+	isOK := func(b *buildapi.Build) bool {
+		return b.Status.Phase == buildapi.BuildPhaseComplete
+	}
+	isFailed := func(b *buildapi.Build) bool {
```

Comment:
this is intentional in CLI. If the build failed, you want to make `oc start-build --wait` return 1 instead of 0, so the script you are running this can make decision based on that status.

In Origin Failed state means the build is done, in start-build --wait it means the command failed.